workspace/apps/pidp/src/app/features/organization-info/pages/endorsements/endorsements.page.ts

```diff
@@ -315,4 +316,25 @@ export class EndorsementsPage
       }),
     );
   }
+
+  public getStatus(endorsementRequestStatus: EndorsementRequestStatus): string {
+    let statusText = 'Requested';
+    if (endorsementRequestStatus === EndorsementRequestStatus.CANCELLED) {
```

Comment:
Rather than else if ladder and checking 3 conditions I suggest defining an object in constants like
const status_text_local_map = {
[EndorsementRequestStatus.CANCELLED]: 'Cancelled',
[EndorsementRequestStatus.DECLINED]: 'EndorsementRequestStatus.DECLINED',
[EndorsementRequestStatus.APPROVED]: 'In progress'
};

Then the current component will import this constant and simply do the following.

` public getStatus(endorsementRequestStatus: EndorsementRequestStatus): string {
             return status_text_local_map.endorsementRequestStatus || ''Requested'';
}
`